### ISSUE TITLE
The training process is fast, but it becomes particularly slow during validation.

### ISSUE THREAD

**Issue by shiyao1999** [Lightning-AI/lightning#19029]
### Bug description

The training speed is fast, approximately 30 minutes per epoch, but it becomes very slow during validation. My training set consists of about 50,000 images, the validation set has around 10,000 images, and I'm using 2 A5000 GPUs for training with a distributed data parallel (ddp) strategy. I'm using PyTorch Lightning version 1.6.0 and Torch version 1.8.1.

### What version are you seeing the problem on?

v1.8

### How to reproduce the bug

```python
class CROHMEDatamodule(pl.LightningDataModule):
    def __init__(
        self,
        train_data: list,
        val_data: list,
        maxImagesize,
        train_batch_size: int = 8,
        eval_batch_size: int = 4,
        num_workers: int = 5,
        scale_aug: bool = False,
    ) -> None:
        super().__init__()
        self.train_data = train_data
        self.val_data = val_data
        self.maxImagesize = maxImagesize
        self.train_batch_size = train_batch_size
        self.eval_batch_size = eval_batch_size
        self.num_workers = num_workers
        self.scale_aug = scale_aug


    def setup(self,stage=None) -> None:
        self.train_dataset = CROHMEDataset(build_dataset(self.train_data, self.train_batch_size, self.maxImagesize),
                                           True, self.scale_aug)
        self.val_dataset = CROHMEDataset(build_dataset(self.val_data, self.eval_batch_size, self.maxImagesize), False,
                                         self.scale_aug)


    def train_dataloader(self):
        return DataLoader(
            self.train_dataset,
            shuffle=True,
            num_workers=self.num_workers,
            collate_fn=collate_fn,
        )

    def val_dataloader(self):
        return DataLoader(
            self.val_dataset,
            shuffle=False,
            num_workers=self.num_workers,
            collate_fn=collate_fn,
        )
```
This is my config:
```
seed_everything: 7
trainer:
  checkpoint_callback: true
  callbacks:
    - class_path: pytorch_lightning.callbacks.LearningRateMonitor
      init_args:
        logging_interval: epoch
    - class_path: pytorch_lightning.callbacks.ModelCheckpoint
      init_args:
        save_top_k: 1
        monitor: val_ExpRate
        mode: max
        filename: '{epoch}-{step}-{val_ExpRate:.4f}'
  gpus: 0, 1, 2, 3
  accelerator: ddp
  check_val_every_n_epoch: 2
  max_epochs: 300
  deterministic: true
model:
  d_model: 256
  # encoder
  growth_rate: 24
  num_layers: 16
  # decoder
  nhead: 8
  num_decoder_layers: 4
  dim_feedforward: 1024
  dropout: 0.3
  dc: 32
  cross_coverage: true
  self_coverage: true
  # beam search
  beam_size: 10
  max_len: 200
  alpha: 1.0
  early_stopping: false
  temperature: 1.0
  # training
  learning_rate: 0.08
  #lr_decrease_rate: 2
  patience: 20
data:
  train_data: ['./data/chinese/train.zip','./data/chinese/train_labels.txt' ]
  val_data: ['./data/chinese/test.zip','./data/chinese/test_labels.txt']
  train_batch_size: 6
  eval_batch_size: 3
  num_workers: 8
  maxImagesize: 500000
  scale_aug: true
```
```
from pytorch_lightning.utilities.cli import LightningCLI
from comer.datamodule import CROHMEDatamodule
from comer.lit_comer import LitCoMER

```
```


### Error messages and logs


![捕获](https://github.com/Lightning-AI/lightning/assets/127834692/3cdd085f-2475-4460-b0b0-0b535035b7d8)
The validation process is exceptionally slow.

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:
                - NVIDIA RTX A5000
                - NVIDIA RTX A5000
        - available:         True
        - version:           11.1
* Lightning:
        - pytorch-lightning: 1.6.0
        - torch:             1.8.1+cu111
        - torchmetrics:      0.5.0
        - torchvision:       0.9.1+cu111
* Packages:
        - absl-py:           0.13.0
        - addict:            2.4.0
        - aiohttp:           3.8.4
        - aiosignal:         1.3.1
        - anyconfig:         0.9.10
        - anyio:             3.3.1
        - appdirs:           1.4.4
        - argon2-cffi:       21.1.0
        - async-timeout:     4.0.2
        - attrs:             21.2.0
        - babel:             2.9.1
        - backcall:          0.2.0
        - black:             20.8b1
        - bleach:            4.1.0
        - brotlipy:          0.7.0
        - cachetools:        4.2.2
        - certifi:           2021.5.30
        - cffi:              1.14.6
        - chardet:           4.0.0
        - charset-normalizer: 3.1.0
        - click:             8.1.3
        - comm:              0.1.2
        - conda:             4.10.3
        - conda-package-handling: 1.7.3
        - cryptography:      3.4.7
        - cycler:            0.10.0
        - debugpy:           1.6.6
        - decorator:         5.1.0
        - defusedxml:        0.7.1
        - docstring-parser:  0.15
        - editdistance:      0.5.3
        - einops:            0.6.0
        - entrypoints:       0.3
        - flake8:            3.9.0
        - frozenlist:        1.3.3
        - fsspec:            2023.3.0
        - future:            0.18.2
        - google-auth:       1.35.0
        - google-auth-oauthlib: 0.4.6
        - grpcio:            1.40.0
        - idna:              2.10
        - imageio:           2.31.5
        - imgaug:            0.4.0
        - ipykernel:         6.21.3
        - ipython:           7.27.0
        - ipython-genutils:  0.2.0
        - ipywidgets:        7.6.5
        - isort:             5.8.0
        - jedi:              0.18.0
        - jinja2:            3.0.1
        - json5:             0.9.6
        - jsonargparse:      3.9.0
        - jsonschema:        3.2.0
        - jupyter:           1.0.0
        - jupyter-client:    7.0.3
        - jupyter-console:   6.6.3
        - jupyter-core:      5.2.0
        - jupyter-server:    1.11.0
        - jupyterlab:        3.1.12
        - jupyterlab-language-pack-zh-cn: 0.0.1.dev0
        - jupyterlab-pygments: 0.1.2
        - jupyterlab-server: 2.8.1
        - jupyterlab-widgets: 1.0.2
        - kiwisolver:        1.3.2
        - markdown:          3.3.4
        - markupsafe:        2.0.1
        - matplotlib:        3.1.2
        - matplotlib-inline: 0.1.3
        - mccabe:            0.6.1
        - mistune:           0.8.4
        - multidict:         6.0.4
        - mypy-extensions:   1.0.0
        - natsort:           8.4.0
        - nbclassic:         0.3.2
        - nbclient:          0.5.4
        - nbconvert:         6.1.0
        - nbformat:          5.1.3
        - nest-asyncio:      1.5.1
        - networkx:          3.1
        - notebook:          6.4.4
        - numpy:             1.21.1
        - oauthlib:          3.1.1
        - opencv-python:     4.1.2.30
        - packaging:         21.0
        - pandocfilters:     1.5.0
        - parso:             0.8.2
        - pathspec:          0.11.1
        - pexpect:           4.8.0
        - pickleshare:       0.7.5
        - pillow:            8.3.2
        - pip:               21.1.3
        - platformdirs:      3.1.1
        - polygon3:          3.0.8
        - prometheus-client: 0.11.0
        - prompt-toolkit:    3.0.38
        - protobuf:          3.18.0
        - psutil:            5.9.4
        - ptyprocess:        0.7.0
        - pyasn1:            0.4.8
        - pyasn1-modules:    0.2.8
        - pyclipper:         1.1.0.post3
        - pycodestyle:       2.7.0
        - pycosat:           0.6.3
        - pycparser:         2.20
        - pydeprecate:       0.3.2
        - pyflakes:          2.3.1
        - pygments:          2.10.0
        - pyopenssl:         20.0.1
        - pyparsing:         2.4.7
        - pyrsistent:        0.18.0
        - pysocks:           1.7.1
        - python-dateutil:   2.8.2
        - pytorch-lightning: 1.6.0
        - pytz:              2021.1
        - pywavelets:        1.4.1
        - pyyaml:            6.0.1
        - pyzmq:             22.3.0
        - qtconsole:         5.4.1
        - qtpy:              2.3.0
        - regex:             2022.10.31
        - requests:          2.25.1
        - requests-oauthlib: 1.3.0
        - requests-unixsocket: 0.2.0
        - rsa:               4.7.2
        - ruamel-yaml-conda: 0.15.100
        - scikit-image:      0.16.2
        - scipy:             1.10.1
        - send2trash:        1.8.0
        - setuptools:        52.0.0.post20210125
        - shapely:           1.6.4.post2
        - six:               1.16.0
        - sniffio:           1.2.0
        - supervisor:        4.2.2
        - tensorboard:       2.6.0
        - tensorboard-data-server: 0.6.1
        - tensorboard-plugin-wit: 1.8.0
        - terminado:         0.12.1
        - testpath:          0.5.0
        - toml:              0.10.2
        - torch:             1.8.1+cu111
        - torchmetrics:      0.5.0
        - torchvision:       0.9.1+cu111
        - tornado:           6.1
        - tqdm:              4.66.1
        - traitlets:         5.9.0
        - typed-ast:         1.5.4
        - typing-extensions: 4.8.0
        - urllib3:           1.26.6
        - wcwidth:           0.2.5
        - webencodings:      0.5.1
        - websocket-client:  1.2.1
        - werkzeug:          2.0.1
        - wheel:             0.36.2
        - widgetsnbextension: 3.5.1
        - yarl:              1.8.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.10
        - release:           5.4.0-107-generic
        - version:           #121-Ubuntu SMP Thu Mar 24 16:04:27 UTC 2022

</details>





### More info

What should I do?

cc @justusschock @awaelchli @borda

**Comment by awaelchli** [comment]
@shiyao1999 Did you try this?
https://lightning.ai/docs/pytorch/1.6.1/advanced/model_parallel.html#when-using-ddp-strategies-set-find-unused-parameters-false

Since the version of lightning you are using is ancient, we probably cant action this, only guess. If nothing helps, I strongly suggest to upgrade to a more recent version.

**Comment by shiyao1999** [comment]
> @shiyao1999 Did you try this? https://lightning.ai/docs/pytorch/1.6.1/advanced/model_parallel.html#when-using-ddp-strategies-set-find-unused-parameters-false
> 
> Since the version of lightning you are using is ancient, we probably cant action this, only guess. If nothing helps, I strongly suggest to upgrade to a more recent version.

I have tried setting find-unused-parameters to False, but it didn't work. I am now updating the version to the latest to see if that helps.

**Comment by shiyao1999** [comment]
> > @shiyao1999 Did you try this? https://lightning.ai/docs/pytorch/1.6.1/advanced/model_parallel.html#when-using-ddp-strategies-set-find-unused-parameters-false
> > Since the version of lightning you are using is ancient, we probably cant action this, only guess. If nothing helps, I strongly suggest to upgrade to a more recent version.
> 
> I have tried setting find-unused-parameters to False, but it didn't work. I am now updating the version to the latest to see if that helps.

I have tried pytorch_lightning==1.3.5, 1.6.0, or 2.1.2. The training is fast, but the validation is very slow. In version 2.1.2, training on 100 data points takes only 26 seconds, but inference on 100 data points requires 20 minutes.

**Comment by awaelchli** [comment]
This is not generally something we observe in our examples. I suspect this is something specific to your use case. Since we don't have your code, we can't help much. But to investigate you should disable as many optional features as possible, for example by setting `Trainer(barebones=True)`.

**Comment by shiyao1999** [comment]
> This is not generally something we observe in our examples. I suspect this is something specific to your use case. Since we don't have your code, we can't help much. But to investigate you should disable as many optional features as possible, for example by setting `Trainer(barebones=True)`.

Thank you very much, my problem has been solved. The reason is that I will perform a beam search during validating and adjust the beam_size to 10. This will require too many calculations. I adjust the beam_size to 3 and the problem is solved.
